src/rpc/client.cpp

```diff
@@ -282,20 +278,14 @@ UniValue RPCConvertNamedValues(const std::string &strMethod, const std::vector<s
     for (const std::string &s: strParams) {
         size_t pos = s.find('=');
         if (pos == std::string::npos) {
-            positional_args.push_back(rpcCvtTable.convert(strMethod, positional_args.size()) ? ParseNonRFCJSONValue(s) : s);
+            positional_args.push_back(rpcCvtTable.ParseIfNonString(s, strMethod, positional_args.size()));
             continue;
         }
 
         std::string name = s.substr(0, pos);
         std::string value = s.substr(pos+1);
```

Comment:
unrelated: I wonder if avoiding a copy (maybe via string_view) would improve performance when large values are passed (like a full hex block)

Reply:
`ParseIfNonString` passes by reference, and in this case we `push_back` an rvalue already anyway so I don't think we're copying anything?

Reply:
I meant that `substr` creates a copy of the string. It may be possible to avoid this by using a string_view, which can then be passed to `ParseIfNonString`

Reply:
Makes sense, I thought you were talking about the highlighted lines - thanks. Proposed in #26612 as I think it's pretty orthogonal to this PR but I like the idea (and I'd been considering simplifying `ParseNonRFCJSONValue` already, which ties in nicely).